scipy/interpolate/tests/test_rgi.py

```diff
@@ -688,6 +689,34 @@ def test_bad_solver(self):
                 (x, y), data, method='slinear',  solver_args={'woof': 42}
             )
 
+    def test_concurrency(self):
+        barrier = threading.Barrier(10)
+        points, values = self._get_sample_4d()
+        sample = np.array([[0.1 , 0.1 , 1.  , 0.9 ],
+                           [0.2 , 0.1 , 0.45, 0.8 ],
+                           [0.5 , 0.5 , 0.5 , 0.5 ]])
+        interp = RegularGridInterpolator(points, values, method="slinear")
+
+        methods = ['slinear', 'cubic']
+        def worker_fn(interp):
+            spline = interp._spline
+            barrier.wait()
+            method = methods[threading.get_ident() % 2]
+            interp(sample, method=method)
+            assert interp._spline is spline
```

Comment:
What does this test, why would `_spline` change?

Reply:
See comment above